doc/primer.rst

```diff
@@ -167,6 +167,190 @@ iterates on each member of the stack.
+These classes actually wrap the equivalent Open CASCADE Technology (OCCT) classes.
+The 9 topological classes are :
+
+1. :class:`~cadquery.Shape`
+2. :class:`~cadquery.Compound`
+3. :class:`~cadquery.CompSolid`
+4. :class:`~cadquery.Solid`
+5. :class:`~cadquery.Shell`
+6. :class:`~cadquery.Face`
+7. :class:`~cadquery.Wire`
+8. :class:`~cadquery.Edge`
+9. :class:`~cadquery.Vertex`
+
+Each class has its own methods to create and/or edit shapes of their respective type. As already explained in :ref:`cadquery_concepts` there is also some kind of hierarchy in the 
+topological classes. A Wire is made of several edges which are themselves made of several vertices. This means you can create geometry from the bottom up and have a lot of control over it.
+
+For example we can create a circular face like so ::
+
+  circle_wire = Wire.makeCircle(10,Vector(0,0,0), Vector(0,0,1))
+  circular_face = Face.makeFromWires(circle_wire, [])
+
+.. note::
+  In CadQuery (and OCCT) all the topological classes are shapes, the :class:`~cadquery.Shape` class is the most abstract topological class. 
+  The topological class inherits :class:`~cadquery.Mixin3D` or :class:`~cadquery.Mixin1D` which provide aditional methods that are shared between the classes that inherits them.
+
+The direct API as its name suggests doesn't provide a parent/children data structure, instead each method call directly returns an object of the specified topological type.
+It is more verbose than the fluent API and more tedious to work with, but as it offer more flexibility (you can work with faces, which is something you can't do in the fluent API)
+it is sometimes more convenient than the fluent API.
+
+The OCCT API
+~~~~~~~~~~~~~~~~~~~~~~
+
+Finally we are discussing about the OCCT API. The OCCT API is the lowest level of CadQuery. The direct API is built upon the OCCT API, where the OCCT API in CadQuery is available through OCP.
+OCP are the Python bindings of the OCCT C++ libraries CadQuery uses. This means you have access to all the OCCT C++ libraries in Python and in CadQuery.
+Working with the OCCT API will give you the maximum flexibility and control over you designs, it is however very verbose and difficult to use. You will need to have a strong 
+knowledge of the different C++ libraries to be able to achieve what you want. To obtain this knowledge the most obvious ways are :
+
+1. Read the direct API source code, since it is build upon the OCCT API it is full of example usage.
+2. Go through the `C++ documentation <https://dev.opencascade.org/doc/overview/html/>`_
+
+.. note::
+  The general way of importing a specific class of the OCCT API is ::
+
+    from OCP.thePackageName import theClassName
+
+  For example if you want to use the class `BRepPrimAPI_MakeBox <https://dev.opencascade.org/doc/refman/html/class_b_rep_prim_a_p_i___make_box.html>`_.
+  You will go by the following ::
+
+    from OCP.BRepPrimAPI import BRepPrimAPI_MakeBox
+
```

Comment:
I think this is just general Python knowledge, not really specific to CQ or OCP.

```suggestion
```

Reply:
What do you think about this suggestion @Jojain ?

Reply:
Well the intent here is to specifically talk about how to import binded c++ classes coming from OCP. I wrote this example because I struggled at first knowing how to import a specific class from OCP.
I my opinion this is much needed, since I couldn't find this information anywhere else.

Reply:
OK, clear